src/lib/mina_net2/libp2p_stream.ml

```diff
@@ -126,6 +126,40 @@ let stream_closed ~logger ~who_closed t =
         ] ;
   `Stream_should_be_released (equal_state FullyOpen t.state)
 
+let max_message_size = 16777216 (* 16 MiB *)
+
+let split_string ~every b =
+  let blen = String.length b in
+  let rec isplit sofar pos =
+    if blen - pos > every then
+      let chunk = String.sub b ~pos ~len:every in
+      isplit (chunk :: sofar) @@ (pos + every)
+    else if blen - pos = 0 then sofar
+    else String.sub ~pos ~len:(blen - pos) b :: sofar
```

Comment:
This creates a 1-length list with a 0-length string for a message of length 0. I assume this is okay/preferable to a 0-length list?